src/qibolab/instruments/bluefors.py

```diff
@@ -0,0 +1,91 @@
+import json
+import re
+import socket
+from datetime import datetime
+
+from qibo.config import log
+
+from qibolab.instruments.abstract import Instrument
+
+
+class TemperatureController(Instrument):
+    """Bluefors temperature controller.
+
+    ```
+    # Example usage
+    if __name__ == "__main__":
+        tc = TemperatureController("XLD1000_Temperature_Controller", "192.168.0.114", 8888)
+        tc.connect()
+        temperature_values = tc.read_data()
+        for temperature_value in temperature_values:
+            print(temperature_value)
+    ```
+    """
+
+    def __init__(self, name: str, address: str, port: int = 8888):
+        """Creation of the controller object.
+
+        Args:
+            name (str): name of the instrument.
+            address (str): IP address of the board sending cryo temperature data.
+            port (int): port of the board sending cryo temperature data.
+        """
+        self.port = port
+        self.client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        super().__init__(name, address)
```

Comment:
```suggestion
        super().__init__(name, address)
        self.port = port
        self.client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

In principle, local settings should take precedence over shared ones. Unless you need to set something first because is then used in the shared code.